peoplehr/types.bal

```diff
@@ -928,3 +928,151 @@ public type ApplicantInformation record {|
     string Email?;
     *Authentication;
 |};
+
+# Represents the Custom Fields
+#
+# + ColumnName - The name of the custom field  
+# + ColumnValue - The value of the custom field
+public type Customfield record {
+    string ColumnName;
+    string ColumnValue;
+};
+
+# Represents Files Details
+#
+# + DocumentName - Name of the document name  
+# + Description - Description of the file  
+# + FileUrl - URL of the file  
+# + Category - File category  
+# + SignatureRequired - Whether a signature is required for the document
+public type File record {
+    string DocumentName; 
+    string Description; 
+    string FileUrl; 
+    string Category; 
+    boolean SignatureRequired;  
+};
+
+# Represents Files Details
+#
+# + DocumentName - Name of the document name  
+# + Description - Description of the file  
+# + File - The file content in Base64 string  
+# + DocumentCategory - File category  
+# + SignatureRequired - Whether a signature is required for the document
+public type NewFile record {
+    string DocumentName; 
+    string Description; 
+    string File; 
+    string DocumentCategory; 
+    boolean SignatureRequired;  
+};
+
+# Represents Employee Screen Details
+#
+# + ScreenId - ScreenId is the unique identifier of the employee custom screen  
+# + ScreenName - ScreenName is the name of the employee custom screen  
+# + TxnId - Field Description  
+# + Customfields - Custom fields details  
+# + Files - Files details
+public type EmployeeScreenDetail record {
+    int ScreenId?;
+    string ScreenName?;
+    int TxnId?;
+    Customfield[] Customfields?;
+    File[] Files?;
+};
+
+# Represents the response of `GetEmployeeScreenDetail` operation  
+#
+# + Result - The response of the operation
+public type EmployeeScreenDetailGetResponse record {|
+    *PeopleHRGenericResponse;
+    EmployeeScreenDetail[]|json Result;
+|};
+
+# Represents request details for retrieving screen details by employee ID
+#
+# + EmployeeId - EmployeeId for employee custom screen
+# + ScreenId -   ScreenId for employee custom screen
+public type ScreenDetailByEmployeeIDRequest record {
+    string EmployeeId;
+    int ScreenId;
```

Comment:
@jdchathuranga we are relying on json-record mapping support coming with Ballerina. This requires field name to be exactly mapping with json keys in the response. We did not find it is a problem as it works in the desired manner and most APIs (almost all) had proper json message keys starting with a simple letter. 

I guess this special case is maybe because they have kind of converted their SOAP API to REST. I can see "Action" is also passed as a value in json types. 

We may evaluate following in a future release. 

```
function fromEmployeeScreenDetailExt(EmployeeScreenDetailExt[] screensExt) returns EmployeeScreenDetail[] {
    return from var { ScreenId: screenId,
                      ScreenName: screenName,
                      TxnId: txnId,
                      Customfields,
                      Files } in screensExt
           let CustomField[]? customFields = Customfields == () ? () : fromCustomfieldsExt(Customfields)
           let File[]? files = Files == () ? () : fromFilesExt(Files)
           select { screenId, screenName, txnId, customFields, files };
}

function fromFilesExt(FileExt[] filesExt) returns File[] {
    return from var { DocumentName: documentName,
                      Description: description,
                      FileUrl: fileUrl,
                      Category: category,
                      SignatureRequired: signatureRequired
                     } in filesExt
           select { documentName, description, fileUrl, category, signatureRequired };
}

function fromCustomfieldsExt(CustomFieldExt[] customFieldsExt) returns CustomField[] {
    return from var { ColumnName: columnName, ColumnValue: columnValue } in customFieldsExt select { columnName, columnValue };
}
```